presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -473,6 +478,23 @@ else if (parameterType.isPrimitive() && !hasNullFlag) {
             }
         }
 
+        private void checkTypeParameters(TypeSignature signature, Method method, ImmutableSet<String> typeParameterNames)
+        {
+            if (typeParameterNames.contains(signature.getBase())) {
+                checkArgument(signature.getParameters().isEmpty(), "Annotated type parameter %s on the method [%s] cannot take parameters.", signature.getBase(), method);
+                return;
+            }
+
+            List<TypeSignatureParameter> parameters = signature.getParameters();
+            for (TypeSignatureParameter parameter : parameters) {
+                Optional<TypeSignature> typeSignature = parameter.getTypeSignatureOrNamedTypeSignature();
+                if (typeSignature.isPresent()) {
+                    // Only check Type and NamedType
```

Comment:
This comment doesn't add much value in my opinion. I would remove it.

It is explaining exactly what line 490 did, without more abstraction or at a higher level.